src/devices/machine/ldv4200hle.h

```diff
@@ -0,0 +1,215 @@
+		CMD_SEARCH               = make_ldv4000_command("SE"),
+		CMD_MULTITRACK_FORWARD   = make_ldv4000_command("JF"),
+		CMD_MULTITRACK_REVERSE   = make_ldv4000_command("JR"),
+		CMD_STOP_MARKER          = make_ldv4000_command("SM"),
+		CMD_FRAME_SET            = make_ldv4000_command("FR"),
+		CMD_TIME_SET             = make_ldv4000_command("TM"),
+		CMD_CHAPTER_SET          = make_ldv4000_command("CH"),
+		CMD_CLEAR                = make_ldv4000_command("CL"),
+		CMD_LEADOUT_SYMBOL       = make_ldv4000_command("LO"),
+		CMD_AUDIO_CTRL           = make_ldv4000_command("AD"),
+		CMD_VIDEO_CTRL           = make_ldv4000_command("VD"),
+		CMD_KEY_LOCK             = make_ldv4000_command("KL"),
+		CMD_DISPLAY_CONTROL      = make_ldv4000_command("DS"),
+		CMD_CLEAR_SCREEN         = make_ldv4000_command("CS"),
+		CMD_PRINT_CHAR           = make_ldv4000_command("PR"),
+		CMD_REQ_FRAME_NUMBER     = make_ldv4000_command("?F"),
+		CMD_REQ_TIME_CODE        = make_ldv4000_command("?T"),
+		CMD_REQ_CHAPTER_NUMBER   = make_ldv4000_command("?C"),
+		CMD_REQ_PLAYER_MODE      = make_ldv4000_command("?P"),
+		CMD_REQ_DISC_STATUS      = make_ldv4000_command("?D"),
+		CMD_REQ_LDP_MODEL        = make_ldv4000_command("?X"),
+		CMD_REQ_PIONEER_DISC_ID  = make_ldv4000_command("?U"),
+		CMD_REQ_STANDARD_DISC_ID = make_ldv4000_command("$Y"),
+		CMD_REQ_TV_SYSTEM        = make_ldv4000_command("?S"),
+		CMD_COMMUNICATION_CTRL   = make_ldv4000_command("CM"),
+		CMD_REQ_CCR_MODE         = make_ldv4000_command("?M"),
+		CMD_REGISTER_A_SET       = make_ldv4000_command("RA"),
+		CMD_REGISTER_B_SET       = make_ldv4000_command("RB"),
+		CMD_REGISTER_C_SET       = make_ldv4000_command("RC"),
+		CMD_REGISTER_D_SET       = make_ldv4000_command("RD"),
+		CMD_REQ_REGISTER_A       = make_ldv4000_command("$A"),
+		CMD_REQ_REGISTER_B       = make_ldv4000_command("$B"),
+		CMD_REQ_REGISTER_C       = make_ldv4000_command("$C"),
+		CMD_REQ_REGISTER_D       = make_ldv4000_command("$D"),
+		CMD_REQ_INPUT_UNIT       = make_ldv4000_command("#I"),
+		CMD_INPUT_NUMBER_WAIT    = make_ldv4000_command("?N")
+	};
+
+	enum player_mode : uint8_t
+	{
+		MODE_PARK,
+		MODE_DOOR_OPEN,
+		MODE_PAUSE,
+		MODE_PLAY,
+		MODE_MS_FORWARD,
+		MODE_MS_REVERSE,
+		MODE_SEARCH,
+		MODE_STILL
+	};
+
+	enum address_mode : uint8_t
+	{
+		ADDRESS_FRAME,
+		ADDRESS_TIME,
+		ADDRESS_CHAPTER
+	};
+
+	enum error_code : uint8_t
+	{
+		ERR_NONE                = 0xff,
+		ERR_COMMUNICATION       = 0,
+		ERR_NOT_AVAILABLE       = 4,
+		ERR_MISSING_ARGUMENT    = 6,
+		ERR_DISC_NOT_LOADED     = 11,
+		ERR_SEARCH              = 12,
+		ERR_DEFOCUS             = 13,
+		ERR_PICTURE_STOP        = 15,
+		ERR_OTHER_INPUT         = 16,
+		ERR_PANIC               = 99
+	};
+
+	void add_command_byte(uint8_t data);
+
+	void queue_reply(const char *reply);
+	void queue_error(error_code err);
+
+	void normalize_command_buffer();
+	void process_command_buffer();
+
+	uint32_t bcd_to_literal(uint32_t bcd);
+	bool is_number(char value);
```

Comment:
It should be possible to make both of these `static` member functions.

Reply:
Done, nice catch.